pygama/dsp/_processors/multi_t_filter.py

```diff
@@ -0,0 +1,169 @@
+    """
+    for i in range(tp_max, 0, -1):
+        if(wf_in[i]>threshold and wf_in[i-1]<threshold):
+            tp_out = i
+            return tp_out
+    tp_out = 0
+    return tp_out
+    
+
+
+@guvectorize(["void(float32[:], float32[:],float32[:], float32, float32[:])",
+              "void(float64[:], float64[:], float64[:], float64, float64[:])",
+              "void(int32[:], int32[:], int32[:], int32, int32[:])",
+              "void(int64[:], int64[:], int64[:], int64, int64[:])"],
+             "(n),(m),(),()->(m)", nopython=True, cache=True)
+def multi_triggering_filter(wf_in, arbitary_list, threshold, delta, tp0_list):
+    """
+    Gets list of indices of the start of leading edges of multiple peaks within a waveform.
+    Is built to handle afterpulses/delayed cross talk and trains of pulses.
+    Parameters
+    ----------
+    wf_in : array-like
+        the array of data within which the list of tp0s will be found
+    arbitraty_list : array-like
+        A fixed length array of the same dimensions that you need the tp0 list to be
+        Used to tell numba what dimension of an array it should expect
+    threshold: scalar 
+        threshold to search for using time_point_thresh
+    delta : scalar
+        the absolute level by which data must vary (in one direction) about an
+        extremum in order for it to be tagged
+    Returns
+    -------
+    tp0_list : array-like
+        Array of fixed length (padded with nans) that hold the indices of
+        the identified initial rise times of peaks in the signal
+        
+    """
+     
+    # initialize arrays, padded with the elements we want
+    
+    tp0_list[:] = np.nan 
+    max_list= np.zeros_like(tp0_list)
+    min_list = np.zeros_like(tp0_list)
+    n_max = np.array([0],dtype=np.int32)
+    n_min = np.array([0],dtype=np.int32)
+    flag = np.array([0],dtype=np.int32)
+
+    # call njit_get_i_local_extrema to get locations of the maxima and minima within a waveform
+    
+    njit_get_i_local_extrema(wf_in, delta,max_list,min_list,n_max,n_min,flag)
+    
+    # Go through the list of maxima, calling njit_time_point_thresh on each maximum (ignoring the nan padding)
+    for i in range(int(n_max[0])):
+        if not np.isnan(max_list[i]): 
+            tp0_list[i] = njit_time_point_thresh(wf_in,float(threshold[0]),int(max_list[i]),tp0_list[i])
+            
+    """ 
+    time_point_thresh has issues with afterpulsing in waveforms that causes  
+    an aferpulse peak's tp0 to be sent to 0 or the same index as the tp0 for the first pulse.
+    This only happens when the relative minimum between the first pulse and 
+    the afterpulse is greater than the threshold. So, we sweep through the array again 
+    to ensure there are no duplicate indices. If there are duplicate indicies caused by a
+    misidentified tp0 of an afterpulse, we replace its index by that of the corresponding minimum
+    found using the njit_get_i_local_extrema function.
+    """
+
```

Comment:
I would suggest turning this next block into a new guvectorized processor called something like `remove_duplicates` or something like that